src/tickcontext.ts

```diff
@@ -271,12 +267,27 @@ export class TickContext extends Tickable {
     return this;
   }
 
-  getStave(): Stave | undefined {
-    throw new RuntimeError('NotImplemented', 'getStave() not implemented.');
-  }
-
-  // eslint-disable-next-line
-  setStave(stave: Stave): this {
-    throw new RuntimeError('NotImplemented', 'setStave() not implemented.');
+  getFormatterMetrics(): FormatterMetrics {
+    return {
+      // The freedom of a tickable is the distance it can move without colliding
+      // with neighboring elements. A formatter can set these values during its
+      // formatting pass, which a different formatter can then use to fine tune.
+      freedom: { left: 0, right: 0 },
+
+      // The simplified rational duration of this tick as a string. It can be
+      // used as an index to a map or hashtable.
+      duration: '',
+
+      // The number of formatting iterations undergone.
+      iterations: 0,
+
+      // The space in pixels allocated by this formatter, along with the mean space
+      // for tickables of this duration, and the deviation from the mean.
+      space: {
+        used: 0,
+        mean: 0,
+        deviation: 0,
+      },
+    };
```

Comment:
This method always returns a dummy object with everything set to 0.

In formatter.ts, Formatter.evaluate() modifies the dummy object, so it's basically a no-op.

Formatter.tune() uses information from getFormatterMetrics(). However, your PR returns a dummy object so tune() gets all zeroes.

I confirmed that if I comment out this line, the buggy test case will look like your PR:
https://github.com/0xfe/vexflow/blob/490970b96aace865669d5d2d44acfeb349bc112f/src/system.ts#L229

So maybe tune() is buggy?